crypto/tls/tls_test.go

```diff
@@ -109,7 +109,7 @@ func TestGetTLSConfig_ClientCerts(t *testing.T) {
 	tlsConfig, err := c.GetTLSConfig()
 	assert.NoError(t, err)
 	assert.Equal(t, false, tlsConfig.InsecureSkipVerify, "make sure we default to not skip verification")
-	assert.Equal(t, 1, len(tlsConfig.Certificates), "ensure a certificate is returned")
+	assert.NotNil(t, tlsConfig.GetClientCertificate, "ensure a get certificate function is returned")
```

Comment:
I suggest also testing what GetClientCertificate returns.
```suggestion
	require.NotNil(t, tlsConfig.GetClientCertificate, "ensure GetClientCertificate is set")
	cert, err := tlsConfig.GetClientCertificate(nil)
	require.NoError(t, err)
	assert.NotNil(t, cert, "ensure GetClientCertificate returns a certificate")
```